### PR TITLE
[RFR] Migrate DisabledInput to use useInput

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

**Comment by Kmaschta** [comment]
Do we still need `<DisabledInput />`?
What's the difference with `<TextInput source="id" disabled />`?

File: packages/ra-ui-materialui/src/input/DisabledInput.js [hunk 1]
```diff
@@ -1,42 +1,47 @@
...
-    />
-);
+}) => {
+    const {
+        id,
+        input: { value },
+    } = useInput({
+        resource,
+        source,
+        type: 'checkbox',
```
Comment (Kmaschta) [inline packages/ra-ui-materialui/src/input/DisabledInput.js]: Why default it to checkbox ? :o

**Comment by djhi** [comment]
> What's the difference with <TextInput source="id" disabled />?

It's not disabled and we don't pass the event handlers (change, blur, focus)

**Comment by djhi** [comment]
PS: I agree with you though. I think having a readonly textinput is weird, especially without a `readonly` nor `disabled` attribute and it's probably not good regarding accessibility.

I suggest we either deprecate it and add a `readonly` attribute. We should also print a message to explain how to migrate from it

**Comment by djhi** [comment]
However, this is not the purpose of this PR. I'm just migrating components from the injection pattern to hooks usage. Let's open an issue about the DisabledInput

**Pull request merged**